packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -67,21 +140,124 @@ public function createIndex(int $domainId, string $index)
+     * @param string $index
+     */
+    public function reindexFromCurrentIndexToNewIndex(int $domainId, string $index)
+    {
+        $indexes = $this->client->indices();
+        $indexName = $this->getVersionedIndexName($domainId, $index);
+        if ($indexes->exists(['index' => $this->getCurrentIndexName($domainId)]) && $indexes->exists(['index' => $indexName])) {
+            $body = [
+                'source' => [
+                    'index' => $this->getCurrentIndexName($domainId),
+                ],
+                'dest' => [
+                    'index' => $indexName,
+                ],
+            ];
+
+            $this->client->reindex([
+                'body' => $body,
+            ]);
+            $this->setting->setForDomain(Setting::ELASTICSEARCH_INDEX, $indexName, $domainId);
+        }
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     */
+    public function createAliasForIndex(int $domainId, string $index)
+    {
+        $indexes = $this->client->indices();
+        $aliasName = $this->getAliasName($domainId, $index);
+        $indexName = $this->getVersionedIndexName($domainId, $index);
+
+        if (!$indexes->existsAlias(['name' => $aliasName, 'index' => $indexName]) && $indexes->exists(['index' => $indexName])) {
+            $indexes->putAlias([
+                'index' => $indexName,
+                'name' => $aliasName,
+            ]);
+        }
+    }
+
+    /**
+     * @param string $aliasName
+     * @param int $domainId
+     */
+    public function deleteNotUsedIndexes(string $aliasName, int $domainId)
+    {
+        $indexes = $this->client->indices();
+
+        foreach ($this->getExistingIndexNamesForAlias($aliasName) as $indexName) {
+            if ($indexName !== $this->getCurrentIndexName($domainId)) {
+                $indexes->deleteAlias([
+                    'index' => $indexName,
+                    'name' => $aliasName,
+                ]);
+
+                $indexes->delete(['index' => $indexName]);
+            }
+        }
+    }
+
+    /**
+     * @param string $aliasName
+     * @return array
+     */
+    public function getExistingIndexNamesForAlias(string $aliasName): array
+    {
+        $existingIndexNames = [];
+        $indexes = $this->client->indices();
+
+        if ($indexes->existsAlias(['name' => $aliasName])) {
+            $aliases = $indexes->getAlias([
+                'name' => $aliasName,
+            ]);
+            foreach (array_keys($aliases) as $indexName) {
+                if ($indexes->exists(['index' => $indexName])) {
+                    $existingIndexNames[] = $indexName;
+                }
+            }
+        }
+
+        return $existingIndexNames;
+    }
+
     /**
      * @param int $domainId
      * @param string $index
      * @return array
      */
-    protected function getDefinition(int $domainId, string $index): array
+    public function getDefinition(int $domainId, string $index): array
```

Comment:
is this backward compatible?

Reply:
you are right. This is a BC break. We agreed on this solution together with @DavidOstrozlik this morning (we were rushing).
@DavidOstrozlik please create a new method that will be public and use this old protected deprecated method